### ISSUE TITLE
Remove .ts files from published npm module

### ISSUE THREAD

**Issue by quantummkv** [Borewit/music-metadata#25]
The published npm module contains both the typescript and javascript files. 
When using the module with tsloader and webpack, it throws about 117 errors.

```
ERROR in ./node_modules/music-metadata/lib/index.ts
Module parse failed: E:\electron\nighthawk\node_modules\music-metadata\lib\index.ts Unexpected token (6:20)
You may need an appropriate loader to handle this file type.
| import common from './common';
| import TagMap, {HeaderType} from './tagmap';
| import EventEmitter = NodeJS.EventEmitter;
| import {ParserFactory} from "./ParserFactory";
| import * as Stream from "stream";
 @ ./src/background/library.ts 3:0-37
 @ ./src/ui/pages.tsx
 @ ./src/ui/shell.tsx
 @ ./src/index.tsx
 @ multi (webpack)-dev-server/client?http://localhost:8080 webpack/hot/only-dev-server react-hot-loader/patch ./src/index.tsx

ERROR in [at-loader] ./node_modules/music-metadata/lib/ParserFactory.ts:168:11
    TS6133: 'warning' is declared but never used.

ERROR in [at-loader] ./node_modules/music-metadata/lib/Windows1292Decoder.ts:23:26
    TS7006: Parameter 'a' implicitly has an 'any' type.

ERROR in [at-loader] ./node_modules/music-metadata/lib/Windows1292Decoder.ts:23:29
    TS7006: Parameter 'min' implicitly has an 'any' type.
```

This occurs due to tsloader trying to load the .ts files instead of .js ones. Setting allowsJs to true does not resolve the errors.

When i manually delete the .ts files while keeping the .js and .d.ts files, the module works perfectly.

This occurs for both music-metadata and strtok3.

**Comment by Borewit** [comment]
I deliberately put the TypeScript files (*.ts) back in the distribution after having lots of issues with other tools like debuggers, not trigger recompile after changes, debugging on JavaScript instead of TypeScript, generating JavaScript files in both the distribution folder and the source folder.

I figured by doing this, it would also make debugging easier as the TypeScript files, with the map files, would allow debuggers to use the original source (TypeScript) instead of the generated JavaScript files.

Hmm... dammit.

**Comment by quantummkv** [comment]
as per https://github.com/TypeStrong/ts-loader/issues/278 , tsloader cannot compile ts from node_modules. And they don't plan to add it in future.

It might be better to bundle it with webpack to distribute it. Anyone with advanced use cases can clone the sources into the project itself.

**Comment by Borewit** [comment]
I hope this one fixes your problem [v0.7.12](https://github.com/Borewit/music-metadata/releases/tag/v0.7.12), please confirm.

**Comment by quantummkv** [comment]
v0.7.13 is working perfectly. Thank you for solving it real quick.
